adserver/management/commands/archive_offers.py

```diff
@@ -0,0 +1,229 @@
+        self.stdout.write(_("Running `wc -l %s`...") % output_file)
+        self.stdout.write(
+            subprocess.check_output(
+                ["wc", "-l", str(output_file)],
+                stderr=subprocess.STDOUT,
+                encoding="utf-8",
+            )
+        )
+
+        self.stdout.write(_("Compressing %s...") % output_file)
+        self.stdout.write(
+            subprocess.check_output(
+                ["gzip", str(output_file)],
+                stderr=subprocess.STDOUT,
+                encoding="utf-8",
+            )
+        )
+
+        # Azure storage automatically stores the md5sum in the ContentMd5 header/property
+        # You can verify it after copying with:
+        #  echo -n MD5HEXSUM | xxd -p -r | base64
+        self.stdout.write(_("MD5 summing %s...") % gzip_output_file)
+        self.stdout.write(
+            subprocess.check_output(
+                ["md5sum", str(gzip_output_file)],
+                stderr=subprocess.STDOUT,
+                encoding="utf-8",
+            )
+        )
+
+        self.stdout.write(self.style.SUCCESS(_("Successfully archived %s.") % day))
+
+    def copy_offer_dumps(self):
+        """Copy offer CSV files to settings.BACKUPS_STORAGE."""
+        if not hasattr(settings, "BACKUPS_STORAGE"):
+            self.stdout.write(
+                self.style.WARNING(
+                    _(
+                        "Skipping copying offers to backups (BACKUPS_STORAGE is not defined)..."
+                    )
+                )
+            )
+            return
+
+        self.stdout.write(_("Copying offer dumps to backups..."))
+
+        copied_file_count = 0
+        storage = get_storage_class(settings.BACKUPS_STORAGE)()
+
+        for filepath in self.output_dir.iterdir():
+            storage_path = self.storage_output_dir + filepath.name
+            self.stdout.write(_("Copying %s to backups...") % filepath)
+            if storage.exists(storage_path):
+                self.stdout.write(
+                    self.style.WARNING(
+                        _("- %s already exists in backups") % storage_path
+                    )
+                )
+            with open(filepath, "rb") as fd:
+                storage.save(storage_path, fd)
+            copied_file_count += 1
+
+        self.stdout.write(
+            self.style.SUCCESS(
+                _("Successfully copied %d archives to backups.") % copied_file_count
+            )
+        )
+
+    def delete_offers(self, start_day, end_day):
+        """Deletes offers from the database (requires them to be copied to settings.BACKUPS_STORAGE)."""
+        if not hasattr(settings, "BACKUPS_STORAGE"):
+            self.stdout.write(
+                self.style.WARNING(
+                    _("Skipping deleting archived offers (backups weren't copied)...")
+                )
+            )
+            return
+
+        self.stdout.write(_("Deleting archived offers..."))
```

Comment:
We should log the dates here to be confident it's doing the right thing. We should probably also do one additional confirmation `if input() == 'y'`, since deleting is scary.

Reply:
It requires a command line option `-d` to delete offers. The default is to *not* do it. Adding an input here means it can never be run automatically and it requires input after potentially many minutes (hours?) of doing the actual archiving. Logging the dates, however, seems reasonable.